3rdParty/iresearch/core/formats/columnstore2.cpp

```diff
@@ -368,6 +360,68 @@ class column_base : public column_reader,
     return *stream_;
   }
 
+ protected:
+  column_header& mutable_header() { return hdr_; }
+  void reset_stream(const index_input* stream) { stream_ = stream; }
+
+  bool allocate_buffered_memory(size_t size) {
+    assert(size < static_cast<size_t>(std::numeric_limits<int64_t>::max()));
+    assert(buffered_memory_accounter_);
+    if (!buffered_memory_accounter_(static_cast<int64_t>(size))) {
+      // use string to have always null-terminated name
+      std::string col_name{"<anonymous>"};
+      if (!name().null()) {
+        col_name = name();
+      }
+      IR_FRMT_WARN(
+          "Failed to allocate memory for buffered column id %d name: %s of "
+          "size " IR_SIZE_T_SPECIFIER,
+          header().id, col_name.c_str(), size);
+      return false;
+    }
+    // should be only one alllocation
+    assert(column_data_.empty());
+    column_data_.resize(size);
```

Comment:
Not sure what the semantics of this `allocate_buffered_memory` function are, but I guess when it is called, we want `size` bytes *more*.
At least the memory accounter function is called like that, but here we are using `resize(size)`, which looks like we would resize the underlying storage to just `size`, and not `column_data_.size() + size`. Please confirm that this is the intended behavior.

Reply:
Column data is expected to be empty (see assert above).  So `allocate_buffered_memory` is intended to be called at most once per lifetime of the column